cypress/support/scenarios/yves/locale-scenario.ts

```diff
@@ -0,0 +1,28 @@
+import { autoWired } from '@utils';
+import { injectable } from 'inversify';
+
+@injectable()
+@autoWired
+export class LocaleScenario {
+  getAvailableLocales = (): Cypress.Chainable => {
+    return cy.get('[data-qa="language-selector"] option').then((options) => {
+      const values = Array.from(options).map((option) => option.textContent?.trim() || '');
+      return Cypress.Promise.resolve(values);
+    });
+  };
+
+  switchLocale = (locale: string): void => {
+    cy.get('[data-qa="language-selector"]').last().select(locale);
```

Comment:
same as above. Any page DOM manipulations should be in in a page class + repository

Reply:
should I duplicate it in each page class that is used?

Reply:
I think in this case, since the language switch is on every Yves page, the methods should be cypress/support/pages/yves/yves-page.ts

Reply:
ah, okay, thank you